File: kit-php/src/Application/Todo/DeleteTodo.php
<?php

declare(strict_types=1);

namespace TodoApp\Application\Todo;

use TodoApp\Domain\Todo\TodoNotFoundException;
use TodoApp\Domain\Todo\TodoRepository;

/** Cas d'usage : supprimer une tâche. */
final class DeleteTodo
{
    public function __construct(private readonly TodoRepository $todos)
    {
    }

    /**
     * @return array{deleted:bool}
     * @throws TodoNotFoundException
     */
    public function execute(string $id): array
    {
        if ($this->todos->find($id) === null) {
            throw TodoNotFoundException::withId($id);
        }

        $this->todos->delete($id);

        return ['deleted' => true];
    }
}

File: kit-fastapi/app/application/todo/delete_todo.py
"""Cas d'usage : supprimer une tâche."""
from __future__ import annotations

from app.domain.todo.exceptions import TodoNotFoundError
from app.domain.todo.repository import TodoRepository


class DeleteTodo:
    def __init__(self, todos: TodoRepository) -> None:
        self._todos = todos

    def execute(self, todo_id: str) -> dict:
        if self._todos.find(todo_id) is None:
            raise TodoNotFoundError.with_id(todo_id)
        self._todos.delete(todo_id)
        return {"deleted": True}
